src/core/memory.d

```diff
@@ -810,6 +810,91 @@ struct GC
     }
 }
 
+/**
+ * Pure variants of C's memory allocation functions `malloc`, `calloc`, and
+ * `realloc` that achieve purity by aborting the program on failure so
+ * they never visibly change errno.
+ *
+ * The functions may terminate the program using `onOutOfMemoryError` or
+ * `assert(0)`. These functions' purity guarantees no longer hold if
+ * the program continues execution after catching AssertError or
+ * OutOfMemoryError.
```

Comment:
I think this comment needs updating.

Reply:
I kept the mention of the possibility so if we later change to using `onOutOfMemoryError` no one will be blindsided.  Do you think it's better to remove it because that's too unlikely?

Reply:
No need to change anything immediately, IMO.  We can wait to see how the review goes.  It just eventually needs to reflect the final implementation.